### PR TITLE
Stage 2 compiler MVP

### PR DESCRIPTION
**Pull request by angustrau**
Now able to compile the simplest program!

### CODE REVIEW

File: Mapping/types.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+class EV3Type(ABC):
+    @staticmethod
+    @abstractmethod
+    def parse(input: str) -> Any:
+        pass
+
+class InPort(EV3Type):
+    @staticmethod
+    def parse(input: str):
```
Comment (glipR) [inline Mapping/types.py:16]: Pretty sure 1.A should go to `in1`, not `inA`.

File: Mapping/utils.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
+import ast
+from collections import OrderedDict
+import copy
+import functools
+import textwrap
+from Utility.memo import memoise
+from .types import EV3Type, Local
+
+methods = {}
+setupCode = OrderedDict()
```
Comment (glipR) [inline Mapping/utils.py:10]: BTW we could just always include all of the setupcode (imports, device dictionaries, etc) and do away with the stateful part of this. It wouldn't affect performance.

Comment (angustrau) [reply Mapping/utils.py:10]: There's now _another_ stateful component, global variable allocation. It's currently very hacky, only works in mapping functions, and relies on setupCode

**Review by glipR — CHANGES_REQUESTED** [review]
All looks good to me! Very nice.

**Review by glipR — APPROVED** [review]

**Pull request merged**